format-delimited/src/main/java/io/cdap/plugin/format/delimited/input/PathTrackingDelimitedInputFormat.java

```diff
@@ -128,4 +150,82 @@ public void close() throws IOException {
       }
     };
   }
+
+  /**
+   * Iterator that provides the splits in the delimited string based on the delimiter. The delimiter
+   * should not contain any quotes. The splitor will behave like this: 1. if there is no quote, it
+   * will behave same as {@link String#split(String)} 2. if there are quotes in the string, the method
+   * will find pairs of quotes, content within each pair of quotes will not get splitted even if there
+   * is delimiter in that. For example, if string is a."b.c"."d.e.f" and delimiter is '.', it will get
+   * split into [a, b.c, d.e.f]. if string is "val1.val2", then it will not get splitted since the '.'
+   * is within pair of quotes. If the delimited string contains odd number of quotes, which mean the
+   * quotes are not closed, an exception will be thrown. The quote within the value will always be
+   * trimed.
+   */
+  @VisibleForTesting
+  static class SplitQuotesIterator extends AbstractIterator<String> {
+    private String delimitedString;
+    private String delimiter;
+    private int index;
+    private boolean endingWithDelimiter = false;
+
+    private SplitQuotesIterator () {}
```

Comment:
I think this breaks the checkstyle. Don't need this one, the other constructor should always be used.